src/Build/FileSystem/IFileSystemAdapter.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Shared.FileSystem;
+
+namespace Microsoft.Build.FileSystem
+{
+    internal class IFileSystemAdapter : MSBuildFileSystemBase
```

Comment:
We have `MSBuildFileSystemAdapter` which adapts IFileSystem to MSBuildFileSystem. Here we need it the other way around. All because we can't break some old Nuget version :)

Reply:
Sigh. That was probably it, but yes, please don't break the old Nuget.